lib/css/CssModulesPlugin.js

```diff
@@ -501,6 +519,21 @@ class CssModulesPlugin {
 					codeGenResult.sources.get("css") ||
 					codeGenResult.sources.get("css-import");
 
+				const moduleSourceCode = moduleSource.source().toString();
+				const publicPathRegex = new RegExp(
+					CssUrlDependency.PUBLIC_PATH_PLACEHOLDER,
+					"g"
+				);
+				moduleSource = new ReplaceSource(moduleSource);
+				let match;
+				while ((match = publicPathRegex.exec(moduleSourceCode))) {
+					/** @type {ReplaceSource} */ (moduleSource).replace(
+						match.index,
+						(match.index += match[0].length - 1),
+						publicPath
+					);
+				}
```

Comment:
Ideally we need to use `ReplaceSource` here for caching, just an idea

Reply:
https://github.com/webpack/webpack-sources?tab=readme-ov-file#replacesource

Reply:
Oh I see, add cache for css renderModule result just like `_moduleFactoryCache` added in js renderModule

Reply:
Yeah, feel free to try it, should work without perf and source maps problems